### PR TITLE
Adds the Auto-Doc Mark IX

### PR DESCRIPTION
**Pull request by ghost**
:cl:
add: Added the Auto-Doc Mark IX, a machine capable of performing most surgeries automatically!
/:cl:

[Video of the autodoc in action](https://streamable.com/9tuk5)

### CODE REVIEW

**Comment by diegoxt45** [comment]
reduce the time the surgery takes and we'l be GOOD

**Comment by diegoxt45** [comment]
also add e-mag functionality

**Comment by ghost** [comment]
@diegoxt45 the more you upgrade it, the faster it goes.

also if you emag it, it'll give the next/current guy a fate worse than death.

**Comment by diegoxt45** [comment]
> @diegoxt45 the more you upgrade it, the faster it goes.
> 
> also if you emag it, it'll give the next/current guy a fate worse than death.

which is...

**Comment by ghost** [comment]
they'll be nuggeted, given severe brain damage, and have their screen shake for a while. also they'll slur.

**Comment by diegoxt45** [comment]
> they'll be nuggeted, given severe brain damage, and have their screen shake for a while. also they'll slur.

sounds like a fun gimmick

**Comment by MaterialisticThings** [comment]
+1

**Comment by karmaisblackandbluepilled** [comment]
+1 can't wait for viv to veto it

**Comment by Majick1** [comment]
+1 for the item itself yeah

**Comment by GuyonBroadway** [comment]
This is something I've wanted for literal years.

Like since 2015.

**Comment by cris5311** [comment]
+1

**Comment by diegoxt45** [comment]
why ins't this merged yet

**Comment by Maroonenahmoron** [comment]
This looks like a fantastic machine to add in, I’ll give this my +1

**Comment by AnthonyJNgo** [comment]
This just sounds unbalanced as hell, but it's popular, so I'm open to being told how it isn't gamebreaking.

**Comment by ghost** [comment]
@AnthonyJNgo 

1. it takes 55k techweb points total to unlock.  
2. if someone finds you in it, and emags it, you're at their mercy.
3. you can't stop the surgery once it's started. this is bad if someone clicks on a bad surgery before you click on a good one.  
4. it can't do brainwashing, organ removal, or stun reduction surgery.

**Comment by yoyobatty** [comment]
Any checks if it gets destroyed in the middle of surgery ?

**Comment by ghost** [comment]
@yoyobatty 
```
/obj/machinery/autodoc/Destroy()
	active_surgery.complete()
	open_machine()
	return ..()
```

yes

**Comment by epicss13gamer** [comment]
Woot woot better autosurgeon +1

**Comment by WarRage333** [comment]
You can do any self surgery in under a minute, its really not overpowered lmao

**Comment by diegoxt45** [comment]
also +1

File: hippiestation/code/game/machinery/autodoc.dm [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+		active_surgery.complete()
+	open_machine()
+	return ..()
+
+/obj/machinery/autodoc/proc/mcdonalds(mob/living/carbon/victim)
+	for(var/obj/item/bodypart/BP in victim.bodyparts)
+		if(BP.body_part != HEAD && BP.body_part != CHEST && BP.dismemberable)
+			playsound(src, 'sound/weapons/circsawhit.ogg', 50, TRUE)
+			BP.drop_limb()
+			victim.emote("scream")
```
Comment (yoyobatty) [inline hippiestation/code/game/machinery/autodoc.dm:148]: Jesus Christ, how horrifying.

File: hippiestation/code/game/machinery/autodoc.dm [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+				))
+
+/obj/machinery/autodoc/MouseDrop_T(mob/target, mob/user)
+	if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK) || !Adjacent(target) || !user.Adjacent(target) || !iscarbon(target))
+		return
+	if(close_machine(target))
+		log_combat(user, target, "inserted", null, "into [src].")
+	add_fingerprint(user)
+
+/obj/machinery/autodoc/emag_act(mob/user)
```
Comment (yoyobatty) [inline hippiestation/code/game/machinery/autodoc.dm:303]: Probably log this proc.

**Review by yoyobatty — CHANGES_REQUESTED** [review]

**Comment by yoyobatty** [comment]
Code looks good, I would like to test this myself to see if I can come with any possible ways to break this.

**Comment by WarRage333** [comment]
What would happen if you get teleported away mid surgery by bluespace dust ingestion or genetic power random teleport

**Review by yoyobatty — APPROVED** [review]

**Pull request merged**